packages/boxed-expression-component/src/table/BeeTable/BeeTableHeader.tsx

```diff
@@ -263,29 +254,46 @@ export function BeeTableHeader<R extends object>({
       ? _.dropRight(reactTableInstance.headerGroups)
       : reactTableInstance.headerGroups;
 
-    const done = new Set<ReactTable.ColumnInstance<R>>();
+    const processedHeaders: Set<String> = new Set();
+
     return headerGroupsToRender.map((headerGroup, headerGroupLevel) => {
       // rowIndex === -1 --> Last headerGroup
       // rowIndex === -2 --> Second to last headerGroup
       // ... and so on
+
       const rowIndex = -(headerGroupsToRender.length - 1 - headerGroupLevel + 1);
 
       const { key, ...props } = { ...headerGroup.getHeaderGroupProps(), style: {} };
       return (
         <tr key={key} {...props}>
-          {headerGroup.headers.map((column, columnIndex) => renderColumn(rowIndex, column, columnIndex, done))}
+          {headerGroup.headers.map((column, columnIndex) => {
+            let lastLevelPlaceholderDepth = 1;
+            let lastLevelPlaceholder = column;
+
+            while (lastLevelPlaceholder.placeholderOf) {
+              lastLevelPlaceholderDepth++;
+              lastLevelPlaceholder = lastLevelPlaceholder.placeholderOf as ReactTable.HeaderGroup<R>;
+              processedHeaders.add(lastLevelPlaceholder.id);
+            }
+
+            return renderColumn(
+              rowIndex,
+              column.placeholderOf ? lastLevelPlaceholder : column,
```

Comment:
@jomarko Minor suggestion: If I correctly understood this logic, you can always set `lastLevelPlaceholder` parameter, without the check. `lastLevelPlaceholder` contains the reference to column `column.placeholderOf` is null. Makes sense?